testapi.pm

```diff
@@ -177,7 +177,31 @@ sub record_soft_failure {
 
     $autotest::current_test->record_soft_failure_result($reason);
     $autotest::current_test->{dents}++;
-    return;
+}
+
+=head2 record_step
+
+    record_step($title, $output [, $result]);
+
+Example:
+
+  record_step('workaround', "we know what we are doing");
+
+Record a generic step result on the current test modules. This is meant for
+informational purposes to be interpreted by a displaying system. For example
+openQA can show a info box as part of the job results details. Use this instead
+of C<record_soft_failure> for example when you do not want to mark the job as a
+softfail. The optional value C<$result> can be 'ok' (default), 'fail',
+'softfail'.
+
+=cut
+
+sub record_step {
+    my ($title, $output, $result) = @_;
+    $result //= 'ok';
+    die "unsupported \$result $result" unless $result =~ /^(ok|fail|softfail)$/;
```

Comment:
Given the recent discussion about possible new result state, I'm for creating helper like is_valid_result.